### ISSUE TITLE
When is viewDidLoad called? (Or, more precisely, what event shows a Morph has been displayed?)

### ISSUE THREAD

**Issue by rickmcgeer** [LivelyKernel/lively.next#608]
I build a Canvas-descended morph as a component, and  the model's initialization instructed a Canvas-based library (ChartJS) to draw a chart.  This required the HTML element, which doesn't exist until the component is instantiated and rendered.  I thought that the `viewDidLoad` method on `ViewModel` was called when this happened, however, it was invoked when the call `part(ChartJSEditorWindow)` was invoked.
I put in some workaround code to check for the id, but I'd like to know the clean way to do this.

**Comment by linusha** [comment]
Yes, the `viewDidLoad` method is called when the View is attached to the model, which is indeed the case when `part` gets called and translates the component definitions to actual morphs. This happens independently  of actually being rendered or not.

How is this component opened? Would overwriting `openInWorld()` work for you? I think some kind of workaround will always be required, as there is no canonical way to **actually ensure** if a morphs has been rendered or nor, but we can definitely think about adding this!
